src/thor/triplegbuilder.cc

```diff
@@ -1202,25 +1203,18 @@ TripLegBuilder::Build(const AttributesController& controller,
       trip_edge->set_end_shape_index(shape.size() - 1);
     }
 
-    // For mapmatching when both start and end point are on the same edge, the duration of the leg
-    // should be set according to the difference of the percentage along the edge
-    //                start_node     end_node
-    // edge start =======0=============0==== edge end
-    //                   |_____________|
-    //                      duration
-    double edge_pct = trim_duration ? end_pct - start_pct : 1.0;
-
     // Set shape attributes
     SetShapeAttributes(controller, tile, edge, mode_costing[static_cast<int>(path_begin->mode)],
-                       shape.begin(), shape.end(), trip_path, origin_second_of_week, edge_pct);
+                       shape.begin(), shape.end(), trip_path, origin_second_of_week,
+                       end_pct - start_pct);
```

Comment:
this was a bug i noticed. we werent doing partial shape attributes for trivial routes